pkgs/tools/networking/ntpsec/default.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv, lib, fetchurl, 
+  wafHook, pkgconfig, gnum4,
+  asciidoc, libxslt, docbook_xsl,
+  bison, openssl_1_1, libbsd,
+  coreutils, pythonPackages, pps-tools,
+  libcap ? null, libseccomp ? null,
+}:
+
+assert stdenv.isLinux -> libcap != null;
+assert stdenv.isLinux -> libseccomp != null;
+
+let
+  python = pythonPackages.python;
+
+  withSeccomp = stdenv.isLinux && (stdenv.isi686 || stdenv.isx86_64);
+in
+
+stdenv.mkDerivation rec {
+  name = "ntpsec-${version}";
+  version = "1.1.4";
+
+  src = fetchurl {
+    url = "https://ftp.ntpsec.org/pub/releases/ntpsec-${version}.tar.gz";
+    sha256 = "1vv5ckjlmaa9p2ri3fnqh804fxn0wl31jkkjn6v6hfw5z69vkwbn";
+  };
+
+  patches = [
+    ./skip-decodenetnum-tests.patch
+  ];
+
+  preBuild = ''
+    substituteInPlace ./ntpclients/ntpleapfetch \
+      --replace cat ${coreutils}/bin/cat \
+      --replace basename ${coreutils}/bin/basename \
+      --replace chmod ${coreutils}/bin/chmod
+  '';
+
+  wafConfigureFlags = [
+    "--sysconfdir=/etc"
+    "--localstatedir=/var"
+    "--refclock=all"
+    "--enable-leap-smear"
+    "--python=${python}/bin/python"
+  ]
+    ++ stdenv.lib.optional withSeccomp "--enable-seccomp";
+
+
+  nativeBuildInputs = [ pkgconfig wafHook gnum4
+    asciidoc docbook_xsl libxslt
+    python pythonPackages.wrapPython
+  ];
+
+  buildInputs = [ libbsd bison openssl_1_1 python ]
+    ++ lib.optional withSeccomp libseccomp
+    ++ lib.optionals stdenv.isLinux [ pps-tools libcap ];
+
+  postFixup = ''
+    wrapPythonPrograms
+  '';
+
+  hardeningEnable = [ "pie" ];
+
+  meta = with stdenv.lib; {
+    homepage = https://www.ntpsec.org/;
+    description = "A secure, hardened, and improved implementation of Network Time Protocol derived from NTP Classic";
+    license = with licenses; [ cc-by-40 bsd2 bsd3 mit ntp ];
+    maintainers = [ maintainers.georgyo ];
+    platforms = platforms.linux;
```

Comment:
If the platform is Linux only, do you have to do the checks and optional expressions for Linux?

Reply:
The package itself can be built for most [unix platforms](https://docs.ntpsec.org/latest/build.html#unix). I think the author is just trying to say it's only verified on linux for now.

Reply:
This is another thing copied from the [ntp package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/networking/ntp/default.nix). Which seemed like a good starting point for NTPsec.

As @jonringer said,  NTPsec is tested on a great many platforms, including macos and freebsd as well as both big endian and little endian systems. The full list of upstream tests can be found here: https://gitlab.com/NTPsec/ntpsec/blob/master/.gitlab-ci.yml

So while it should work fine on other systems,  I only have the ability to test on linux.